# EDA Inicial: AI Developer Productivity Dataset

**Fecha**: 5 de febrero de 2026  
**Dataset**: `data/ai_dev_productivity.csv`

## Objetivos
1. Confirmar las dimensiones del dataset (500 filas esperadas)
2. Verificar valores faltantes/nulos
3. Calcular rango de cada variable (min, max, promedio)

In [33]:
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', None)
pd.set_option('display.precision', 2)

print(f"Pandas version: {pd.__version__}")
print(f"NumPy version: {np.__version__}")

Pandas version: 3.0.0
NumPy version: 2.4.2


In [34]:
df = pd.read_csv('../data/ai_dev_productivity.csv')
print("Dataset cargado correctamente.")

Dataset cargado correctamente.


---
## 1. ¿Cuántos datos tenemos?

In [35]:
filas, columnas = df.shape
print(f"Filas: {filas}")
print(f"Columnas: {columnas}")

print(f"\nColumnas del dataset:")
for i, col in enumerate(df.columns, 1):
    print(f"  {i}. {col}")

Filas: 500
Columnas: 9

Columnas del dataset:
  1. hours_coding
  2. coffee_intake_mg
  3. distractions
  4. sleep_hours
  5. commits
  6. bugs_reported
  7. ai_usage_hours
  8. cognitive_load
  9. task_success


In [36]:
df.head(10)

,hours_coding,coffee_intake_mg,distractions,sleep_hours,commits,bugs_reported,ai_usage_hours,cognitive_load,task_success
0,5.99,600,1,5.8,2,1,0.71,5.4,1
1,4.72,568,2,6.9,5,3,1.75,4.7,1
2,6.30,560,1,8.9,2,0,2.27,2.2,1
3,8.05,600,7,6.3,9,5,1.40,5.9,0
4,4.53,421,6,6.9,4,0,1.26,6.3,1
5,4.53,429,1,7.1,5,0,3.06,3.9,1
6,8.16,600,1,8.3,6,0,0.30,2.2,1
7,6.53,600,4,3.6,9,3,1.47,9.1,0
8,4.06,409,5,6.1,6,2,2.43,7.0,0
9,6.09,567,5,7.3,7,0,2.11,5.1,1


---
## 2. ¿Hay valores faltantes/nulos?

In [37]:
print("Valores nulos por columna:")
nulos = df.isnull().sum()
print(nulos)

total_nulos = nulos.sum()
print(f"\nTotal de valores nulos en el dataset: {total_nulos}")

if total_nulos == 0:
    print("\nEl dataset esta completo, no hay valores faltantes.")
else:
    print(f"\nSe encontraron {total_nulos} valores faltantes.")
    print("\nPorcentaje de nulos por columna:")
    print((nulos / len(df) * 100).round(2))

Valores nulos por columna:
hours_coding        0
coffee_intake_mg    0
distractions        0
sleep_hours         0
commits             0
bugs_reported       0
ai_usage_hours      0
cognitive_load      0
task_success        0
dtype: int64

Total de valores nulos en el dataset: 0

El dataset esta completo, no hay valores faltantes.


---
## 3. ¿Qué rango tiene cada variable? (min, max, promedio)

In [38]:
df.describe()

,hours_coding,coffee_intake_mg,distractions,sleep_hours,commits,bugs_reported,ai_usage_hours,cognitive_load,task_success
count,500.00,500.00,500.00,500.00,500.00,500.00,500.00,500.00,500.00
mean,5.02,463.19,2.98,6.98,4.61,0.86,1.51,4.50,0.61
std,1.95,142.33,1.68,1.46,2.70,1.10,1.09,1.87,0.49
min,0.00,6.00,0.00,3.00,0.00,0.00,0.00,1.00,0.00
25%,3.60,369.50,2.00,6.10,3.00,0.00,0.70,3.18,0.00
50%,5.03,500.50,3.00,6.95,5.00,0.00,1.26,4.40,1.00
75%,6.27,600.00,4.00,7.90,6.00,2.00,2.07,5.80,1.00
max,12.00,600.00,8.00,10.00,13.00,5.00,6.36,10.00,1.00


In [39]:
resumen = pd.DataFrame({
    'Min': df.min(),
    'Max': df.max(),
    'Promedio': df.mean(),
    'Mediana': df.median(),
    'Rango': df.max() - df.min()
})

print("Resumen de rangos por variable:")
resumen

Resumen de rangos por variable:


,Min,Max,Promedio,Mediana,Rango
hours_coding,0.0,12.00,5.02,5.03,12.00
coffee_intake_mg,6.0,600.00,463.19,500.50,594.00
distractions,0.0,8.00,2.98,3.00,8.00
sleep_hours,3.0,10.00,6.98,6.95,7.00
commits,0.0,13.00,4.61,5.00,13.00
bugs_reported,0.0,5.00,0.86,0.00,5.00
ai_usage_hours,0.0,6.36,1.51,1.26,6.36
cognitive_load,1.0,10.00,4.50,4.40,9.00
task_success,0.0,1.00,0.61,1.00,1.00


In [ ]:
print("Tipos de datos:")
print(df.dtypes)
print(f"\nUso de memoria: {df.memory_usage(deep=True).sum() / 1024:.2f} KB")

Tipos de datos:
hours_coding        float64
coffee_intake_mg      int64
distractions          int64
sleep_hours         float64
commits               int64
bugs_reported         int64
ai_usage_hours      float64
cognitive_load      float64
task_success          int64
dtype: object

Uso de memoria: 35.29 KB


---
## Resumen

### Hallazgos:
1. **Dimensiones**: 500 filas x 9 columnas (confirmado)
2. **Valores nulos**: 0 en todas las columnas, dataset completo
3. **Rangos**:
   - `hours_coding`: 0 - 12 (promedio 5.02)
   - `coffee_intake_mg`: 6 - 600 (promedio 463.19)
   - `distractions`: 0 - 8 (promedio 2.98)
   - `sleep_hours`: 3 - 10 (promedio 6.98)
   - `commits`: 0 - 13 (promedio 4.61)
   - `bugs_reported`: 0 - 5 (promedio 0.86)
   - `ai_usage_hours`: 0 - 6.36 (promedio 1.51)
   - `cognitive_load`: 1 - 10 (promedio 4.50)
   - `task_success`: 0 o 1 (61% de exito)

### Proximos pasos:
- Analisis de correlacion entre variables
- Visualizacion de distribuciones
- Explorar relacion entre AI usage y task success